x-pack/plugins/task_manager/task_manager.ts

```diff
@@ -52,59 +53,78 @@ export class TaskManager {
-      const store = new TaskStore({
-        callCluster: server.plugins.elasticsearch.getCluster('admin').callWithInternalUser,
-        index: config.get('xpack.task_manager.index'),
-        maxAttempts: config.get('xpack.task_manager.max_attempts'),
-        supportedTypes: Object.keys(this.definitions),
-      });
-      const pool = new TaskPool({
-        logger,
-        maxWorkers: this.maxWorkers,
-      });
-      const createRunner = (instance: ConcreteTaskInstance) =>
-        new TaskManagerRunner({
-          logger,
-          kbnServer,
-          instance,
-          store,
-          definition: this.definitions[instance.taskType],
-          beforeRun: this.middleware.beforeRun,
-        });
-      const poller = new TaskPoller({
+    const store = new TaskStore({
+      callCluster: server.plugins.elasticsearch.getCluster('admin').callWithInternalUser,
+      index: config.get('xpack.task_manager.index'),
+      maxAttempts: config.get('xpack.task_manager.max_attempts'),
+      supportedTypes: Object.keys(this.definitions),
+    });
+    const pool = new TaskPool({
+      logger,
+      maxWorkers: this.maxWorkers,
+    });
+    const createRunner = (instance: ConcreteTaskInstance) =>
+      new TaskManagerRunner({
         logger,
-        pollInterval: config.get('xpack.task_manager.poll_interval'),
-        work() {
-          return fillPool(pool.run, store.fetchAvailableTasks, createRunner);
-        },
+        kbnServer,
+        instance,
+        store,
+        definitions: this.definitions,
+        beforeRun: this.middleware.beforeRun,
       });
+    const poller = new TaskPoller({
+      logger,
+      pollInterval: config.get('xpack.task_manager.poll_interval'),
+      work() {
+        return fillPool(pool.run, store.fetchAvailableTasks, createRunner);
+      },
+    });
 
-      bindToElasticSearchStatus(server.plugins.elasticsearch, logger, poller, store);
+    this.logger = logger;
+    this.store = store;
+    this.poller = poller;
 
-      this.store = store;
-      this.poller = poller;
+    kbnServer.afterPluginsInit(async () => {
       this.isInitialized = true;
+      store.addSupportedTypes(Object.keys(this.definitions));
+      await store.init();
+      await poller.start();
+      // schedule all the static tasks
+      Object.keys(this.definitions).forEach((k: string) => {
+        const taskDef = this.definitions[k];
+
+        if (taskDef.static) {
+          taskDef.static.forEach(async (t: TaskInstance) => {
+            t.taskType = k;
+            await this.schedule(t);
+          });
+        }
+      });
     });
   }
 
   /**
    * Method for allowing consumers to register task definitions into the system.
    * @param taskDefinitions - The Kibana task definitions dictionary
    */
-  public registerTaskDefinitions(taskDefinitions: TaskDictionary<TaskDefinition>) {
+  public async registerTaskDefinitions(taskDefinitions: TaskDictionary<TaskDefinition>) {
```

Comment:
is this `async` a stray addition?

Reply:
Yes it was left over from one expirement or another.